sema4ai/src/sema4ai_code/robo/collect_actions_ast.py

```diff
@@ -105,33 +174,75 @@ class ActionInfoTypedDict(TypedDict):
     kind: str
 
 
-def _make_action_info(
-    uri: str, node: ast_module.FunctionDef, kind: str
-) -> ActionInfoTypedDict:
+class DatasourceInfoTypedDict(ActionInfoTypedDict):
+    engine: str
+    model_name: str | None
+    created_table: str | None
+
+
+def _get_ast_node_range(
+    node: ast_module.FunctionDef | ast_module.Call,
+) -> RangeTypedDict:
     coldelta = 4
 
-    return {
-        "range": {
-            "start": {"line": node.lineno, "character": node.col_offset + coldelta},
+    if isinstance(node, ast_module.Call):
+        func_node = node.func
+        return {
+            "start": {
+                "line": func_node.lineno,
+                "character": func_node.col_offset,
+            },
             "end": {
-                "line": node.lineno,
-                "character": node.col_offset + coldelta + len(node.name),
+                "line": node.end_lineno or func_node.lineno,
```

Comment:
I don't really get this one: since we just want the name coordinates, why isn't it directly `func_node.end_lineno` / `func_node.end_col_offset`?

Reply:
It is because of linting. It seems that node.end_lineno is `int | None` while we always expect a number. In our case it will probably never be None, or will be if there is a mistake (e.g: function not closed) so then we default to the start line.